### ISSUE TITLE
Forcing "enable Atom-TypeScript in JavaScript" is apparently unintuitive

### ISSUE THREAD

**Issue by lierdakil** [atom-ide-community/atom-ide-javascript#14]
This is prompted by https://github.com/TypeStrong/atom-typescript/issues/1566

I suggest maybe configure Atom-TypeScript once after installation or track if the option was disabled by the user and set a flag to not auto-enable it. A configuration option that seems "stuck" without an apparent reason can be pretty annoying.

**Comment by aminya** [comment]
I can add notifications to the settings that we modify in this package.

**Comment by Zearin** [comment]
Agreed.  I want the features of this package only for JS.  My personal preference is to avoid TypeScript, and this package is flooding my editor with TypeScript-related messages.

**Comment by GyozaGuy** [comment]
I'd also like to request the ability to turn off TypeScript-specific features. I don't use TypeScript but otherwise love the features of this package.

**Comment by aminya** [comment]
Just to clarify all the features (autocomplete, datatips, signature help, outline, diagnostics, etc) except a few (Eslint) come from Typescript. 
Disabling Typescript means disabling all the features. This the same for all the IDEs like VsCode. JavaScript without Typescript almost has no IDE capabilities.

Could you exactly state what feature (e.g suggestion for installing @types) do you like to disable, so I can provide a config for that.

**Comment by GyozaGuy** [comment]
Sorry, I should be more clear. Needing some of those enabled makes sense. The main things I wish I could disable include:

- @types suggestions (they seem to have disappeared for me today, I'm not sure why)
- UI additions such as TypeScript version, project, etc... in the status bar (minor annoyance, but clutters things up a bit)
- Autocomplete (I don't use it, and it overrides some shortcuts I've gotten used to in Atom, such as typing `lo` and hitting "tab" to get `console.log();`)
- Outline (minor since I can just keep it closed, but disabling it may free up resources? I don't know how these packages work internally)

While using this package I've noticed Atom gets very sluggish with some of my larger files open, which is why I think disabling some features may help. I've also noticed I have three outline panels now: the TypeScript "semantic view", one with colors, and a colorless one with a filter bar at the top. The TypeScript one only pops up if I specifically enable "semantic view", but the other two have the same shortcut so it gets a bit confusing. I thought maybe one came from a previous package I already had, but the only outline package I have installed is `atom-ide-outline`. Since I can't find where the other one comes from, I can't change its shortcut.

After thinking about this, I guess all I really use is diagnostics and hyperclicking. I don't use the other features, so maybe instead I should use different packages rather than asking for this one to be changed. 😅 I do appreciate all of your work on this though!

**Comment by aminya** [comment]
From your message, it seems you have `ide-ui` installed although it is deprecated. Please disable that package to speed up Atom. @lierdakil and I are working on consolidating Semantics View into Outline. Currently, Semantics View has some performance issues, and it is really heavy (even it takes CPU even if you disable it).

With that being said:
- I am fine with providing for disabling the `@types` package. To compensate for lack of @types, I need to implement a feature that automatically installs @types (e.g. @types/node). This will fix #4 as well.
- I agree with removing the UI additions. I think a unified status bar icon (which supports all the languages) instead of multiple icons that take space would be more useful.
- For autocomplete, you should provide a higher priority for your snippets, so they show first.

**Comment by GyozaGuy** [comment]
Thanks for the tip about `atom-ide-ui`, disabling that got rid of the extra outline and seems to have improved performance. I like the idea of the unified status bar icon. And for the autocomplete, I haven't added any snippets, that was just default behavior in Atom that I've grown used to. But I suppose I can add some snippets to get around that easily enough.

Thanks for your consideration!
